driver-core/src/main/com/mongodb/spi/dns/InetAddressResolverProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spi.dns;
+
+/**
+ * Service-provider class for {@link InetAddressResolver}.
+ *
+ * <p> A resolver provider is a factory for custom implementations of {@linkplain
+ * InetAddressResolver InetAddress resolvers}. A resolver defines operations for
+ * looking up (resolving) host names.
+ *
+ * <p>A resolver provider is a concrete implementation of this interface that has a
+ * zero-argument constructor and implements the abstract methods specified below.
+ *
+ * <p>The driver discovers implementations of this interface via {@link java.util.ServiceLoader}.
```

Comment:
The first of the two paragraphs is both excessive and incomplete (it misses the ["The type must be public and must not be an inner class"](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/util/ServiceLoader.html) requirement). The same appears to be the case for the original line in the documentation of `java.net.spi.InetAddressResolverProvider`. I propose to leave only the second paragraph.

Reply:
I'm not sure which paragraph you want to delete.  You say the first of the two paragraphs, but seems like you're referring to the second paragraph (not including the one-line summary).  Are you asking to delete the paragraph starting with "A resolver provider is a concrete implementation"?  If so, why not fix the paragraph to make it complete?  What's excessive about it?

Reply:
> A resolver provider is a concrete implementation of this interface that has a zero-argument constructor and implements the abstract methods specified below.

Is an inclompete duplicate of

> The driver discovers implementations of this interface via {@link java.util.ServiceLoader}.

The documentation of `ServiceLoader` specifies all the requirements, and we, in my opinion, should not duplicate them.

Reply:
Done, and also done in `DnsClientProvider`